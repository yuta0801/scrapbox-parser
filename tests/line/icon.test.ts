import { parse } from '../../src'

describe('icon', () => {
  it('Simple root icon', () => {
    expect('[/icons/+1.icon]').toMatchSnapshotWhenParsing({ hasTitle: false })
  })

  it('Simple relative icon', () => {
    expect('[me.icon]').toMatchSnapshotWhenParsing({ hasTitle: false })
  })

  it('Multiple icons', () => {
    expect('[me.icon*3]').toMatchSnapshotWhenParsing({ hasTitle: false })
  })

  it('Icon and internal link on same line', () => {
    expect('[Internal link][me.icon]').toMatchSnapshotWhenParsing({
      hasTitle: false
    })
  })

  it('Each multiple icon must be different Object', () => {
    const [block] = parse('[me.icon*2]', { hasTitle: false })
    if (block.type !== 'line') {
      throw new Error('fail')
    }

    const [icon1, icon2] = block.nodes
    expect(icon1 === icon2).toBe(false)
  })

  it('Icon in decoration', () => {
    expect('[** [me.icon]]').toMatchSnapshotWhenParsing({
      hasTitle: false
    })
  })
})
