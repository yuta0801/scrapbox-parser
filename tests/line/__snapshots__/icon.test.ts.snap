// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon Icon and internal link on same line: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "content": "",
        "href": "Internal link",
        "pathType": "relative",
        "type": "link",
      },
      Object {
        "path": "me",
        "pathType": "relative",
        "type": "icon",
      },
    ],
    "type": "line",
  },
]
`;

exports[`icon Icon in decoration: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "decos": Array [
          "*-2",
        ],
        "nodes": Array [
          Object {
            "content": "",
            "href": "me.icon",
            "pathType": "relative",
            "type": "link",
          },
        ],
        "type": "decoration",
      },
    ],
    "type": "line",
  },
]
`;

exports[`icon Multiple icons: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "path": "me",
        "pathType": "relative",
        "type": "icon",
      },
      Object {
        "path": "me",
        "pathType": "relative",
        "type": "icon",
      },
      Object {
        "path": "me",
        "pathType": "relative",
        "type": "icon",
      },
    ],
    "type": "line",
  },
]
`;

exports[`icon Simple relative icon: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "path": "me",
        "pathType": "relative",
        "type": "icon",
      },
    ],
    "type": "line",
  },
]
`;

exports[`icon Simple root icon: toMatchSnapshotWhenParsing 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "path": "/icons/+1",
        "pathType": "root",
        "type": "icon",
      },
    ],
    "type": "line",
  },
]
`;
